gap/generic/SnAnUnknownDegree.gi

```diff
@@ -37,12 +111,35 @@ RECOG.ThreeCycleCandidatesConstants := function(eps, N)
         B := Int(Ceil(13 * Log(Float(N)) * Log(3 / Float(eps)))),
         T := Int(Ceil(3 * Log(3 / Float(eps)))),
         C := Int(Ceil(Float(3 * N * ~.T / 5))),
-        logInt2N := LogInt(N, 2)
+        logInt2N := LogInt(N, 2),
+        K := 5, # this is an arbitrary number, max number for batch of conjugates per involution
+        L := 10, # this is an arbitrary number, max number for batch of involutions
     );
 end;
 
+# Return false if c can not be a three cycle. This uses cheap tests to
+# determine this. It is based on the Magma function heuristicThreeCycleTest.
+RECOG.HeuristicThreeCycleTest := function(ri, c, logInt2N, R)
+    local r, y, yTo5, k;
+    c := StripMemory(c);
+    if not isone(ri)(c ^ 3) then
+        return false;
+    fi;
+    for k in [1 .. logInt2N + 1] do
+        r := R[k];
+        # c * c ^ r is a product of two three-cycles, so it should have order
+        # 1, 2, 3 or 5.
+        y := c * c ^ r;
+        yTo5 := y ^ 5;
+        if not isone(ri)(yTo5) and not isone(ri)(yTo5 * y) then
```

Comment:
Computing $y^5 = $(y^2)^2\cdot y$ require three multiplications, and then $y^5\cdot y$ a fourth one. So either 3 or 4 multiplications

Wouldn't it then be even better to do something like this, which requires either 1, 2 or 3 multiplications ?
```suggestion
        y2 := y^2;
        if not isone(ri)(y2) then
          y3 := y2 * y;
          if not isone(ri)(y3) and not isone(ri)(y2 * y3) then
```

Reply:
yes, your suggestion is definitely better